src/NumberAdder.java

```diff
@@ -0,0 +1,48 @@
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Scanner;
+
+public class NumberAdder {
+    public static void main(String[] args) {
+        List<Double> numbers = readNumbersFromUser();
+        printSum(numbers);
+    }
+
+    private static List<Double> readNumbersFromUser() {
+        List<Double> numbers = new ArrayList<>();
+        Scanner input = new Scanner(System.in);
+        double number = 0;
+        do {
+            System.out.println("Podaj liczbę (liczba ujemna kończy wprowadzanie):");
+            boolean exception = true;
+            while (exception) {
```

Comment:
I don't like that you have two loops if this could be done with only one. Two loops make code more complicated than it should be.